content/blog/2025-12-16-facade-layout-compiler.md

```diff
@@ -0,0 +1,409 @@
++++
+title = "Facade Layout Compiler (# 618)"
+[extra]
+bio = """
+  I-Ting Tsai is a CS PhD student at Cornell University.
+"""
+[[extra.authors]]
+name = "I-Ting Tsai"
++++
+
+# Experience Report: Facade Layout Compiler
+
+[Source Code](https://github.com/itingtsai/Facade_Layout_Compiler)
+
+## What was the goal?
+
+The goal of this project was to design and implement a Facade Layout Compiler—a domain-specific compiler that transforms textual descriptions of building facades into verified, renderable outputs. The input language describes a facade as a symbolic grid of architectural elements (windows, doors, chimneys, spacing, and empty cells), and the compiler produces:
+
+**Verified layouts:** The compiler performs static semantic analysis to enforce architectural constraints such as door placement rules, chimney connectivity, and optional window spacing requirements.
+
+**Renderable outputs:** For valid programs, the compiler generates both SVG visualizations and structured JSON representations suitable for downstream processing or generative design workflows.
+
+This project directly mirrors a traditional compiler pipeline—lexical analysis, parsing, semantic analysis, and code generation—but operates in a visually interpretable domain. The choice of architectural facades as the target domain was intentional: it provides immediate visual feedback for correctness, making it easier to reason about both the compiler's behavior and the quality of its output.
+
+## Alignment with Proposal
+
+My original proposal outlined three key deliverables:
+
+1. **Custom DSL for facade specification:** Implemented with support for row declarations, repeat expressions, compact syntax, comments, and rule declarations
+2. **Static rule enforcement:** Distinct semantic checks with error/warning classification
+3. **Empirical validation through testing:** pass a comprehensive test suite
+
+The scope evolved slightly during implementation. More details are described in the sections below.
+
+---
+
+## What did you do?
+
+The complete implementation is available in [facade_compiler.py](https://github.com/itingtsai/Facade_Layout_Compiler/blob/main/facade_compiler.py)
```

Comment:
```suggestion
The complete implementation is available in [facade_compiler.py](https://github.com/itingtsai/Facade_Layout_Compiler/blob/main/facade_compiler.py).
```

But I'm not sure you need both this and the other link to your repo at the top of the post.